src/Covenant/ASG.hs

```diff
@@ -390,6 +407,12 @@ pattern Thunk i <- ThunkInternal i
 pattern Cata :: Ref -> Ref -> ValNodeInfo
 pattern Cata algebraRef valRef <- CataInternal algebraRef valRef
 
+-- | Inject (zero or more) fields into a data constructor
+--
+-- @since 1.2.0
+pattern DataConstructor :: TyName -> ConstructorName -> Vector Ref -> ValNodeInfo
+pattern DataConstructor tyName ctorName fields <- DataConstructorInternal tyName ctorName fields
+
 {-# COMPLETE Lit, App, Thunk, Cata #-}
```

Comment:
This pragma should be extended so that `DataConstructor` is matched before a match is considered exhaustive.